README.md

```diff
@@ -1,2 +1,127 @@
+
+    **Example:** Launched using the Tower CLI
+
+    ```console
+    tw launch sage-bionetworks-workflows/nf-synindex --params-file=./params.yml
+    ```
+
+6. Explore the parent folder on Synapse.
+
+    **Example:** Synapse files and folders under `syn26601236`
+
+    ```text
+    syn26601270  foobar_1.trimming_report.txt
+    syn26601272  foobar_2.trimming_report.txt
+    syn26601252  fastqc/
+    syn26601268    foobar_1_val_1_fastqc.html
+    syn26601267    foobar_1_val_1_fastqc.zip
+    syn26601271    foobar_2_val_2_fastqc.html
+    syn26601269    foobar_2_val_2_fastqc.zip
+    ```
+
+7. Explore the output file mapping the S3 objects and the corresponding Synapse file IDs.
+
+    **Example:** Downloaded from `s3://example-bucket/outputs/synindex/under-syn26601236/file_ids.csv`
+
+    ```text
+    s3://example-bucket/outputs/foobar_1.trimming_report.txt,syn26601270
+    s3://example-bucket/outputs/foobar_2.trimming_report.txt,syn26601272
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.html,syn26601268
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.zip,syn26601267
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.html,syn26601271
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.zip,syn26601269
+    ```
+
+## Authentication
+
+Downloading files from Synapse requires the workflow to be authenticated. The workflow currently supports two authentication methods:
+
+- **(Preferred)** Create a secret called `SYNAPSE_AUTH_TOKEN` containing a Synapse personal access token using the [Nextflow CLI](https://nextflow.io/docs/latest/secrets.html) or [Nextflow Tower](https://help.tower.nf/latest/secrets/overview/).
+- Provide a Synapse configuration file containing a personal access token (see example above) to the `synapse_config` parameter. This method is best used if Nextflow/Tower secrets aren't supported on your platform.
+
+You can generate a personal access token using [this dashboard](https://www.synapse.org/#!PersonalAccessTokens:).
+
+## Parameters
+
+Check out the [Quickstart](#quickstart) section for example parameter values.
+
+- **`s3_prefix`**: An S3 prefix containing a set of files that need to be indexed in Synapse. Typically, this corresponds to the value given to the `outdir` parameter from an nf-core workflow run.
+
+- **`parent_id`**: The Synapse ID of a Synapse folder that will contain the indexed files and the associated folder structure.
+
+- **`synapse_config`**: (Optional) A [Synapse configuration file](https://python-docs.synapse.org/build/html/Credentials.html#use-synapseconfig) containing authentication credentials. A minimal example is included in the [Quickstart](#quickstart) section.
+
+## Known Limitations
+
+- The workflow doesn't annotate the Synapse files it creates with any metadata or provenance.
+- The workflow will index all of the objects under the given S3 prefix. There is no way to filter or skip the indexing of certain objects.
```

Comment:
```suggestion
- The workflow will index all of the objects under the given S3 prefix. There is no way to filter or skip the indexing of certain objects. Make sure that your `synapse_config` file is not located within a directory that will be indexed on Synapse!
```

Reply:
Great suggestion! I even created an issue (#6) to build in some safeguard in the workflow itself.